### PR TITLE
Switch to Apollo async timing

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: include/RAJA/policy/apollo_multi/kernel/kernel_impl.hpp [hunk 1]
```diff
@@ -29,41 +30,30 @@ struct PostLaunchKernel {
...
-    callback_t *cbdata =
-        reinterpret_cast<callback_t *>(region->callback_pool->get());
-    context->isDoneCallback =
-        RAJA::apollo::ApolloCallbackHelper::isDoneCallback;
-    context->callback_arg = cbdata;
-
-    //std::cout << "PreLaunch CudaKernelAsync\n";
-    // TODO: how to handle multiple streams?
-    cudaEventRecord(cbdata->start, 0);
+    context->timer = Apollo::Timer::create<Apollo::Timer::CudaAsync>();
```
Comment (ggeorgakoudis) [inline include/RAJA/policy/apollo_multi/kernel/kernel_impl.hpp:41]: Timer is not currently under the Apollo name (Apollo::Timer -> Timer). Do you think it's better to include it?

**Review by ggeorgakoudis — COMMENTED** [review]
LGTM, let's decide on including or not Timer under the Apollo class/namespace. Could you try also a run with the simple RAJA test program?

File: include/RAJA/policy/apollo_multi/kernel/kernel_impl.hpp (hunk 1 above)
Comment (davidbeckingsale) [reply include/RAJA/policy/apollo_multi/kernel/kernel_impl.hpp:41]: I think it should be in the Apollo namespace for now.

Comment (ggeorgakoudis) [reply include/RAJA/policy/apollo_multi/kernel/kernel_impl.hpp:41]: Makes sense. I made the change to Apollo to match what RAJA expects.

**Review by ggeorgakoudis — APPROVED** [review]

**Pull request merged**